front/src/modules/timetableItem/components/Timetable/utils.ts

```diff
@@ -100,3 +100,18 @@ export const getTrainCategoryClassName = (
   }
   return null;
 };
+
+// TODO: Reason received when a pathfinding failed. Remove this when issue #12772 is resolved.
+export const isValidPathfinding = (summaryTrain: SimulationSummary | undefined) => {
+  if (!summaryTrain) return false;
+  if (summaryTrain && 'invalidReason' in summaryTrain) {
```

Comment:
You know now that summaryTrain is defined
```suggestion
  if ('invalidReason' in summaryTrain) {
```

Reply:
Done